diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SortRule.kt

```diff
@@ -0,0 +1,109 @@
+
+class SortRule : Rule("sort-rule") {
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.isClassEnum() && node.hasChildOfType(CLASS_BODY))
+            sortEnum(node.findChildByType(CLASS_BODY)!!)
+        if (node.isCompanionObject() &&  node.hasChildOfType(CLASS_BODY))
+            sortProperty(node.findChildByType(CLASS_BODY)!!)
+    }
+
+    private fun sortProperty(node: ASTNode) {
+        val propertyList = node.getChildren(null)
+                .filter { it.elementType == PROPERTY }
+                .filter { it.hasChildOfType(MODIFIER_LIST) }
+                .filter { it.findChildByType(MODIFIER_LIST)!!.hasChildOfType(CONST_KEYWORD) }
+        if (propertyList.size <= 1)
+            return
+        val orderListOfList = createOrderListOfList(propertyList)
+        val sortedListOfList = mutableListOf<MutableList<ASTNode>>()
+        orderListOfList.forEach { nodesList ->
+            val sortList = nodesList.toMutableList()
+            sortList.sortBy { it.findChildByType(IDENTIFIER)!!.text }
+            sortedListOfList.add(sortList)
+        }
+        if (orderListOfList != sortedListOfList) {
+            SORT_RULE.warnAndFix(configRules, emitWarn, isFixMode,
+                    "Constant properties inside companion object order is incorrect", node.startOffset) {
+                sortedListOfList.forEachIndexed { index, mutableList ->
+                    swapSortNodes(mutableList, orderListOfList[index], node)
+                }
+            }
+        }
+    }
+
+    private fun swapSortNodes(sortList: List<ASTNode>, nonSortList: List<ASTNode>, node: ASTNode) {
+        val nodeBefore: ASTNode? = nonSortList.last().treeNext
+        node.removeRange(nonSortList.first(), nonSortList.last().treeNext)
+        sortList.forEachIndexed { nodeIndex, astNode ->
+            if (nodeIndex != 0)
+                node.addChild(PsiWhiteSpaceImpl("\n"), nodeBefore)
+            node.addChild(astNode, nodeBefore)
+        }
+    }
+
+    private fun createOrderListOfList(propertyList: List<ASTNode>): MutableList<MutableList<ASTNode>> {
+        val orderListOfList = mutableListOf<MutableList<ASTNode>>()
+        var oneOrderList = mutableListOf(propertyList.first())
+        propertyList.forEachIndexed { index, astNode ->
+            if (index != propertyList.size - 1) {
+                if (astNode.treeNext.treeNext == propertyList[index + 1]) {
```

Comment:
```suggestion
                if (astNode.nextSibling { it.elementType == PROPERTY } == propertyList[index + 1]) {
```
I think it's more readable and reliable than hardcoded two steps.